lib/practice_exercises.rb

```diff
@@ -1,13 +1,49 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.downcase.gsub(/\W/,'')
+  # string = string.split('')
+
+  reverse_string = string.split('')
+  i = 0
+  j = reverse_string.length - 1
+  while i < j
+    temp = reverse_string[i]
+    reverse_string[i] = reverse_string[j]
+    reverse_string[j] = temp
+    i += 1
+    j -= 1
+  end
+  x = 0
+  string.each_char do |letter|
+
+    if letter != reverse_string[x]
+      return false
+    end
+
+    x += 1
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(1)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
How do you think this is O(1) time complexity?  `sort!` for example is O(n log n), so...  You also have a loop that goes the length of the smallest string.  

Otherwise this works.